pkg/cmd/admin/policy/psp_review.go

```diff
@@ -0,0 +1,302 @@
+				Template: *o.PodTemplateSpec,
+			},
+		}
+		response, err = o.selfSubjectReviewClient.PodSecurityPolicySelfSubjectReviews(o.Namespace).Create(podSecurityPolicySelfSubjectReview)
+		if err != nil {
+			return err
+		}
+	}
+	return o.PrintObject(response, out)
+}
+
+func (o *subjectReviewOptions) Complete(f *clientcmd.Factory, args []string, cmd *cobra.Command) error {
+	if len(o.User) > 0 && len(o.ServiceAccount) > 0 {
+		return fmt.Errorf("--user and --serviceaccount are mutually exclusive")
+	}
+	namespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	o.Namespace = namespace
+	oclient, _, err := f.Clients()
+	if err != nil {
+		return fmt.Errorf("unable to obtain client: %v", err)
+	}
+	o.subjectReviewClient = oclient
+	o.selfSubjectReviewClient = oclient
+
+	o.Namespace = namespace
+	mapper, typer := f.Object()
+	infos, err := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), kapi.Codecs.UniversalDecoder()).
+		NamespaceParam(namespace).
+		FilenameParam(enforceNamespace, &resource.FilenameOptions{Recursive: false, Filenames: []string{o.Filename}}).
+		SingleResourceType().
+		Do().
+		Infos()
+
+	if err != nil {
+		return fmt.Errorf("unable to find a resource: %v", err)
+	}
+	if len(infos) != 1 {
+		return fmt.Errorf("expected a resource for path: %s", o.Filename)
+	}
+	o.PodTemplateSpec, err = returnPodTemplateSpec(infos[0].Object)
+	if err != nil {
+		return fmt.Errorf("unable to find podTemplateSpec from %s: %v", o.Filename, err)
+	}
+
+	o.PrintObject = func(c *cobra.Command, m meta.RESTMapper) func(runtime.Object, io.Writer) error {
+		return func(obj runtime.Object, out io.Writer) error {
+			return f.PrintObject(c, m, obj, out)
+		}
+	}(cmd, mapper)
+
+	return nil
+}
+
+func NewCmdSubjectReview(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	o := &subjectReviewOptions{}
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s", name),
+		Long:    subjectReviewLong,
+		Short:   "Check whether a user or a ServiceAccount can create a Pod.",
+		Example: fmt.Sprintf(subjectReviewExamples, fullName, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := o.Complete(f, args, cmd); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(cmd, err.Error()))
+			}
+			err := o.Run(out)
+			kcmdutil.CheckErr(err)
+		},
+	}
+
+	cmd.Flags().StringVarP(&o.User, "user", "u", o.User, "User")
+	cmd.Flags().StringSliceVarP(&o.Groups, "groups", "g", o.Groups, "List of groups, comma separated")
+	cmd.Flags().StringVarP(&o.Filename, "filename", "f", o.Filename, "Filename containing resource")
+	cmd.Flags().StringVarP(&o.ServiceAccount, "serviceaccount", "z", o.ServiceAccount, "service account in the current namespace to use as a user")
+
+	kcmdutil.AddPrinterFlags(cmd)
+
+	return cmd
+}
+
+// Related to PodSecurityPolicyReview
+var (
+	reviewLong = templates.LongDesc(`Checks which ServiceAccount can create a Pod.
```

Comment:
Move to separate file, please. And like mentioned before, try to keep the flow similar to how other commands are.